scripts/local_integration_test.sh

```diff
@@ -34,26 +34,29 @@ if [[ -z "$imageUnderTest" ]]; then
 fi
 
 # build the test app
-pushd ${testAppDir}
-mvn clean install -DskipTests --batch-mode
-popd
+if [ ! -d $deployDir ]; then
+  echo "Deploy dir $deployDir does not exist. Please build the test application before running\
+ this test."
+  exit 1
+fi
 
 # build app container locally
 pushd $deployDir
-export STAGING_IMAGE=$imageUnderTest
-envsubst < Dockerfile.in > Dockerfile
+# escape special characters in docker image name
+STAGING_IMAGE=$(echo $imageUnderTest | sed -e 's/\//\\\//g')
+sed -e "s/FROM .*/FROM $STAGING_IMAGE/" Dockerfile.in > Dockerfile
 echo "Building app container..."
-docker build -t $APP_IMAGE . || gcloud docker -- build -t $APP_IMAGE .
+docker build -t $APP_IMAGE . || docker -- build -t $APP_IMAGE .
```

Comment:
I guess one option is to add `gcloud` only if the Cloud SDK installation is detected.